Modal.js

```diff
@@ -0,0 +1,199 @@
+// Creates and appends the components for a bulma modal user msg
+export function createModal(id, msg, prompt_title = undefined, type = 'default') {
+
+  /*
+  Modal element tree
+
+  modal
+    |---> background
+    |---> card 
+            |---> header
+            |       |---> title (optional)
+            |---> body
+            |       |---> message
+            |       |---> input (optional)
+            |---> footer
+                    |---> controls
+                            |---> positive (btn)
+                            |---> negative (btn)
+  */
+
+  const modal = document.createElement('div')
+  modal.classList.add('modal', 'is-active')
+  modal.setAttribute('id', id)
+  document.body.append(modal)
+
+  const background = document.createElement('div')
+  background.classList.add('modal-background')
+  modal.append(background)
+
+  const card = document.createElement('div')
+  card.classList.add('modal-card')
+  modal.append(card)
+
+  const header = document.createElement('header')
+  header.classList.add('modal-card-head', 'py-4')
+  card.append(header)
+
+  const body = document.createElement('section')
+  body.classList.add('modal-card-body', 'py-4')
+  prompt_title || type === 'req-input' ? card.append(body) : undefined
+
+  const footer = document.createElement('footer')
+  footer.classList.add('modal-card-foot', 'py-2')
+  card.append(footer)
+
+  const title = document.createElement('p')
+  prompt_title ? title.classList.add('modal-card-title') : title.classList.add('modal-card-title', 'is-size-6')
+  prompt_title ? title.innerText = prompt_title : title.innerText = msg
+  header.append(title)
+
+  const message = document.createElement('p') 
+  message.classList.add('label', 'is-size-6')
+  prompt_title ? message.innerText = msg : undefined
+  body.append(message)
+
+  if (type === 'req-input') {
+    const input = document.createElement('input')
+    input.setAttribute('type', 'text')
+    input.classList.add('input', 'is-focused')
+    body.append(input)
+  }
+
+  const controls = document.createElement('div')
+  controls.classList.add('buttons', 'is-flex', 'is-justify-content-end', 'is-flex-grow-1')
+  footer.append(controls)
+
+  const button_positive = document.createElement('button')
+  button_positive.classList.add('button')
+  button_positive.setAttribute('id', `${id}_submit`)
+  button_positive.innerText = 'OK'
+  controls.append(button_positive)
+
+  const button_negative = document.createElement('button')
+  button_negative.classList.add('button', 'exit')
+  button_negative.setAttribute('id', `${id}_cancel`)
+  button_negative.innerText = 'Cancel'
+  controls.append(button_negative)
+
+  
+  // Add a click event on various child elements to delete the parent modal
+  const altCloseModal = (document.querySelectorAll('.modal-background, .modal-close, .modal-card-head .delete, .modal-card-foot, .exit') || []).forEach(($close) => {
```

Comment:
@Geo-Bold is there a particular reason you're using `|| []` here? If the query selector doesn't match anything it will return an empty array by default.

Reply:
@anthonyisensee No reason. I imported as is from Bulma's modal function library.